tools/codeowners-utils/Azure.Sdk.Tools.CodeownersUtils/Parsing/CodeownersEntryPathComparer.cs

```diff
@@ -0,0 +1,260 @@
+            return 0;
+        }
+
+        /// <summary>
+        /// Handle simple wildcard cases where one path ends with * and matches the other path.
+        /// </summary>
+        private bool HandleSimpleWildcardCase(CodeownersEntry x, CodeownersEntry y, string pathX, string pathY, out int result)
+        {
+            result = 0;
+            bool xHasWildcard = x.ContainsWildcard;
+            bool yHasWildcard = y.ContainsWildcard;
+
+            // Simple check: if one path ends with * and matches the other path, prioritize the wildcard
+            if (xHasWildcard && !yHasWildcard && pathX.EndsWith("*") && pathX.TrimEnd('*') == pathY)
+            {
+                result = -1; // x (with wildcard) comes before y (without wildcard)
+                return true;
+            }
+            if (!xHasWildcard && yHasWildcard && pathY.EndsWith("*") && pathY.TrimEnd('*') == pathX)
+            {
+                result = 1; // y (with wildcard) comes before x (without wildcard)
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Compare two CodeownersEntry objects by their service labels.
+        /// </summary>
+        private int CompareByServiceLabel(CodeownersEntry x, CodeownersEntry y)
+        {
+            // Get the primary service context for each entry (service label first, then PR label)
+            string primaryLabelX = GetPrimaryServiceContext(x);
+            string primaryLabelY = GetPrimaryServiceContext(y);
+
+            // If both have primary labels, compare them alphabetically
+            if (!string.IsNullOrEmpty(primaryLabelX) && !string.IsNullOrEmpty(primaryLabelY))
+            {
+                return string.Compare(primaryLabelX, primaryLabelY, StringComparison.OrdinalIgnoreCase);
+            }
+
+            // If only one has a primary label, prioritize the one with the label
+            if (!string.IsNullOrEmpty(primaryLabelX) && string.IsNullOrEmpty(primaryLabelY))
+            {
+                return -1; // x has primary label, y doesn't - x comes first
+            }
+            if (string.IsNullOrEmpty(primaryLabelX) && !string.IsNullOrEmpty(primaryLabelY))
+            {
+                return 1; // y has primary label, x doesn't - y comes first
+            }
+
+            // If neither has labels, they're equal at this level
+            return 0;
+        }
+
+        /// <summary>
+        /// Get the primary service context for an entry. This prioritizes ServiceLabel but falls back to the first PRLabel.
+        /// This ensures entries with combined PR labels like "AI Model Inference %AI Projects" are grouped with 
+        /// individual "AI Model Inference" entries. For hierarchical services like "Communication - Call Automation",
+        /// this extracts the base service name "Communication" for grouping.
+        /// </summary>
+        private string GetPrimaryServiceContext(CodeownersEntry entry)
```

Comment:
This can be static if it doesn't depend on anything in a CodeownersEntryPathComparer instance.